packages/AlgernonTests.package/ALGTestAction.class/instance/testWindowOperations.st

```diff
@@ -0,0 +1,14 @@
+testing
+testWindowOperations
+
+	| instance |
+	instance := ALGMockedWindowOperationsType new.
+	instance
+		closeAllButWorkspaces;
+		closeAllWindows;
+		closeWithoutChanges;
+		collapseAllWindows.
+	self assert: 3 equals: instance mockedDockingBar calls keys size.
```

Comment:
The current coverage report states that your newly introduced indirections for the docking bar/the world in the WindowType and the WindowOperationsType aren't covered. 

I question if that is still the case if we use `dockingBar` instead of `mockedDockingBar` here. That won't make a difference in behaviour as you overwrote the dockingBar shorthand in the MockedWindowOperationsType with this indirection, right? 😄 
If this will lead to a coverage of the `dockingBar` then the same could be applied to the `world` shorthand here and the `dockingBar` one in the WindowType :)

Reply:
Well, I didn't think of that :laughing: 

I wouldn't change `mockedDockingBar` to just `dockingBar` here, because it is not what wanted to express. The test doesn't _really_ care about docking bars or in which way the window operations get them, it just wants to check up on the mock object. I think if we changed this to `dockingBar`, we would also run into the method of the mocked object and not into the one that is currently not covered.

I think we could add a test that verifies the interface of whatever the `dockingBar` and `world` methods return. Maybe get the object and then assert that it can understand the messages that we need it to understand (`closeAllWindows` etc.). This way, we would have the coverage and the test would be meaningful (maybe we change the docking bar at some point? don't know if that is possible :D)

What do you think?

Reply:
I see your point. Didn't thought about whether that would be really meaningful. Was more like an approach based on pure coverage 😄 

Anyways, the 2 latest commits got us back over 99% regardless :D